radixdlt-core/radixdlt/src/main/java/com/radixdlt/middleware2/network/MempoolAddMessage.java

```diff
@@ -64,36 +64,36 @@
 
 package com.radixdlt.middleware2.network;
 
-import com.radixdlt.atom.Txn;
-
-import java.util.List;
-import java.util.Objects;
-import java.util.stream.Collectors;
-
-import org.radix.network.messaging.Message;
-
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.radixdlt.atom.Txn;
 import com.radixdlt.serialization.DsonOutput;
 import com.radixdlt.serialization.DsonOutput.Output;
 import com.radixdlt.serialization.SerializerId2;
+import org.radix.network.messaging.Message;
+
+import java.util.List;
+import java.util.Objects;
 
 @SerializerId2("message.mempool.add")
 public final class MempoolAddMessage extends Message {
 	@JsonProperty("txns")
 	@DsonOutput(Output.ALL)
 	private final List<byte[]> txns;
 
-	MempoolAddMessage() {
-		// Serializer only
-		this.txns = null;
+	@JsonCreator
+	public MempoolAddMessage(
+		@JsonProperty(value = "txns", required = true) List<byte[]> txns
+	) {
+		this.txns = Objects.requireNonNull(txns);
```

Comment:
If nulls can be injected by cbor then we should check for nulls here as well.

Reply:
done

Reply:
I checked that using methods toDson and fromDson in the Serialization class keep nulls inside a list.